### PR TITLE
sl-173: Start stepper tone of voice

### PR DESCRIPTION
**Pull request by jaroslawtt**
image
with icon
image

### CODE REVIEW

File: frontend/src/packages/sites/libs/enums/enums.ts [hunk 1]
```diff
@@ -1 +1,2 @@
 export { SitesApiPath } from 'shared/build/index.js';
+export { SiteToneType } from 'shared/build/index.js';
```
Comment (what1s1ove) [inline frontend/src/packages/sites/libs/enums/enums.ts]: merge imports

File: shared/src/packages/sites/libs/validation-schemas/site-create.validation-schema.ts [hunk 2]
```diff
@@ -5,6 +5,7 @@ import { type SiteCreateRequestDto } from '~/packages/sites/libs/types/types.js'
 const siteCreate = joi.object<SiteCreateRequestDto, true>({
   name: joi.string().required(),
   industry: joi.string().required(),
+  tone: joi.string().required(),
```
Comment (what1s1ove) [inline shared/src/packages/sites/libs/validation-schemas/site-create.validation-schema.ts]: should be only values from an enum

File: shared/tsconfig.json [hunk 3]
```diff
@@ -12,6 +12,7 @@
       "~/*": ["./src/*"]
     },
     "outDir": "build",
-    "declaration": true
+    "declaration": true,
+    "skipLibCheck": true
```
Comment (what1s1ove) [inline shared/tsconfig.json]: move it

File: shared/src/packages/sites/libs/validation-schemas/site-create-step-tone.validation-schema.ts [hunk 4]
```diff
@@ -0,0 +1,18 @@
+import joi from 'joi';
+
+import { type SiteCreateStepTone } from '~/packages/sites/libs/types/types.js';
+
+import { SiteToneType, SiteValidationMessage } from '../enums/enums.js';
+
+const siteCreateStepTone = joi.object<SiteCreateStepTone>({
```
Comment (what1s1ove) [inline shared/src/packages/sites/libs/validation-schemas/site-create-step-tone.validation-schema.ts]:
```suggestion
const siteCreateStepTone = joi.object<SiteCreateStepTone, true>({
```

File: shared/src/packages/sites/libs/types/site-create-request-dto.type.ts [hunk 5]
```diff
@@ -1,6 +1,10 @@
+import { type ValueOf } from '~/libs/types/value-of.type';
```
Comment (what1s1ove) [inline shared/src/packages/sites/libs/types/site-create-request-dto.type.ts]: import

File: shared/src/packages/sites/libs/types/site-create-request-dto.type.ts [hunk 6]
```diff
@@ -1,6 +1,10 @@
...
+import { type SiteToneType } from '~/packages/sites/libs/enums/enums.js';
```
Comment (what1s1ove) [inline shared/src/packages/sites/libs/types/site-create-request-dto.type.ts]: import
```suggestion
import { type SiteToneType } from '~/packages/sites/sites.js';
```

File: frontend/src/pages/start/libs/constants.ts [hunk 7]
```diff
@@ -1,10 +1,12 @@
+import { SiteToneType } from '~/packages/sites/libs/enums/enums.js';
```
Comment (what1s1ove) [inline frontend/src/pages/start/libs/constants.ts]:
```suggestion
import { SiteToneType } from '~/packages/sites/sites.js';
```

File: frontend/src/pages/start/libs/components/voice-tone-form/voice-tone-form.tsx [hunk 8]
```diff
@@ -0,0 +1,51 @@
+import { Button, Select } from '~/libs/components/components.js';
+import { useAppForm, useCallback } from '~/libs/hooks/hooks.js';
+import { type SiteCreateStepTone } from '~/packages/sites/libs/types/types.js';
+import { siteCreateStepToneValidationSchema } from '~/packages/sites/libs/validation-schemas/validation-schemas.js';
```
Comment (what1s1ove) [inline frontend/src/pages/start/libs/components/voice-tone-form/voice-tone-form.tsx]:
```suggestion
import { siteCreateStepToneValidationSchema } from '~/packages/sites/libs/validation-schemas/validation-schemas.js';
```
```suggestion
import { siteCreateStepToneValidationSchema } from '~/packages/sites/sites.js';
```

File: frontend/src/pages/start/libs/components/voice-tone-form/voice-tone-form.tsx [hunk 9]
```diff
@@ -0,0 +1,51 @@
+import { Button, Select } from '~/libs/components/components.js';
+import { useAppForm, useCallback } from '~/libs/hooks/hooks.js';
+import { type SiteCreateStepTone } from '~/packages/sites/libs/types/types.js';
```
Comment (what1s1ove) [inline frontend/src/pages/start/libs/components/voice-tone-form/voice-tone-form.tsx]: same

File: frontend/src/pages/start/libs/components/voice-tone-form/voice-tone-form.tsx [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+
+type Properties = {
+  onSubmit: (sitePayload: Partial<SiteCreateStepTone>) => void;
+};
+
+const VoiceToneForm: React.FC<Properties> = ({ onSubmit }: Properties) => {
+  const { control, errors, handleSubmit } = useAppForm({
+    defaultValues: defaultPayload,
+    validationSchema: siteCreateStepToneValidationSchema,
+    mode: 'onSubmit',
```
Comment (what1s1ove) [inline frontend/src/pages/start/libs/components/voice-tone-form/voice-tone-form.tsx]:
```suggestion
```
it by default

File: frontend/src/pages/start/libs/components/voice-tone-form/libs/constants.ts [hunk 11]
```diff
@@ -0,0 +1,14 @@
+import { type SelectOption } from '~/libs/types/types.js';
+import { SiteToneType } from '~/packages/sites/libs/enums/enums.js';
+import { type SiteCreateStepTone } from '~/packages/sites/libs/types/types.js';
+
+const options: SelectOption<string | number>[] = [
+  { value: SiteToneType.OFFICIAL, label: 'Official' },
+  { value: SiteToneType.NOT_OFFICIAL, label: 'Not official' },
+];
```
Comment (what1s1ove) [inline frontend/src/pages/start/libs/components/voice-tone-form/libs/constants.ts]:
```suggestion
const options: SelectOption<string | number>[] = Object.values(SiteToneType).map((it) => ({
  value: it,
  label: siteToneTypeToReadable[it]
}))
```

**Review by what1s1ove — CHANGES_REQUESTED** [review]

File: frontend/src/pages/start/libs/components/voice-tone-form/libs/constants.ts [hunk 12]
```diff
@@ -0,0 +1,20 @@
+import { type SelectOption } from '~/libs/types/types.js';
+import {
+  type SiteCreateStepTone,
+  SiteToneType,
+} from '~/packages/sites/sites.js';
+
+import { SiteToneTypeToReadable } from './site-tone-type-to-readable.enum.js';
+
+const OPTIONS: SelectOption<string | number>[] = Object.values(
```
Comment (andanf-e) [inline frontend/src/pages/start/libs/components/voice-tone-form/libs/constants.ts]: Looks like it's not `string | number` - you actually know the type of values passed (`SiteToneType` vals)

File: frontend/src/pages/start/libs/components/voice-tone-form/libs/constants.ts [hunk 13]
```diff
@@ -0,0 +1,20 @@
...
+import { SiteToneTypeToReadable } from './site-tone-type-to-readable.enum.js';
+
+const OPTIONS: SelectOption<ValueOf<typeof SiteToneType>>[] = Object.values(
+  SiteToneType,
+).map((it) => ({
+  value: it,
+  label: SiteToneTypeToReadable[it],
+}));
+
+const defaultPayload: SiteCreateStepTone = {
```
Comment (what1s1ove) [inline frontend/src/pages/start/libs/components/voice-tone-form/libs/constants.ts]:
```suggestion
const DEFAULT_FORM_PAYLOAD: SiteCreateStepTone = {
```

File: frontend/src/pages/start/libs/components/voice-tone-form/libs/site-tone-type-to-readable.enum.ts [hunk 14]
```diff
@@ -0,0 +1,8 @@
+import { SiteToneType } from '~/packages/sites/libs/enums/enums.js';
+
+const SiteToneTypeToReadable = {
```
Comment (what1s1ove) [inline frontend/src/pages/start/libs/components/voice-tone-form/libs/site-tone-type-to-readable.enum.ts]:
```suggestion
const siteToneTypeToReadable = {
```

File: frontend/src/pages/start/libs/components/voice-tone-form/libs/site-tone-type-to-readable.enum.ts [hunk 15]
```diff
@@ -0,0 +1,8 @@
+import { SiteToneType } from '~/packages/sites/libs/enums/enums.js';
```
Comment (what1s1ove) [inline frontend/src/pages/start/libs/components/voice-tone-form/libs/site-tone-type-to-readable.enum.ts]: move to 
`frontend/src/pages/start/libs/components/voice-tone-form/libs/maps/site-tone-type-to-readable.map.ts`

File: frontend/src/pages/start/libs/components/voice-tone-form/libs/constants.ts [hunk 16]
```diff
@@ -0,0 +1,20 @@
+import { type SelectOption, type ValueOf } from '~/libs/types/types.js';
+import {
+  type SiteCreateStepTone,
+  SiteToneType,
+} from '~/packages/sites/sites.js';
+
+import { SiteToneTypeToReadable } from './site-tone-type-to-readable.enum.js';
```
Comment (what1s1ove) [inline frontend/src/pages/start/libs/components/voice-tone-form/libs/constants.ts]:
```suggestion
import { SiteToneTypeToReadable } from './maps/maps.js';
```

**Review by what1s1ove — CHANGES_REQUESTED** [review]

**Review by what1s1ove — APPROVED** [review]

**Pull request merged**